qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1420,6 +1422,37 @@ def collect_runs(self, namelist):
                     group_list.append(tuple(group))
         return set(group_list)
 
+    def collect_1q_runs(self):
+        """Return a set of non-conditional runs of 1q "op" nodes."""
+        group_list = []
+        # Iterate through the nodes of self in topological order
+        # and form tuples containing sequences of gates
+        # on the same qubit(s).
+        topo_ops = list(self.topological_op_nodes())
+        nodes_seen = dict(zip(topo_ops, [False] * len(topo_ops)))
```

Comment:
Would this be faster as an initially empty `set`?

Reply:
Yeah, it probably would be definitely faster to construct. It also avoids a boolean comparison ontop of the set lookup. Not a ton of overhead there, but still something.